diff --git a/subprojects/gvc/meson.build b/subprojects/gvc/meson.build
index a1a2af5..f3dab57 100644
--- a/subprojects/gvc/meson.build
+++ b/subprojects/gvc/meson.build
@@ -55,7 +55,7 @@ libgvc_no_gir_sources = [
 libgvc_deps = [
   dependency('gio-2.0'),
   dependency('gobject-2.0'),
-  dependency('libpulse', version: '>= 12.99.3'),
+  dependency('libpulse', version: '>= 5'),
   dependency('libpulse-mainloop-glib')
 ]
 
