diff --git a/drivers/devfreq/governor_passive.c b/drivers/devfreq/governor_passive.c
index bc88ded..10d2a39 100644
--- a/drivers/devfreq/governor_passive.c
+++ b/drivers/devfreq/governor_passive.c
@@ -293,7 +293,7 @@ static int cpufreq_passive_register(struct devfreq_passive_data **p_data)
 	unsigned int cpu;
 	int ret;
 
-	get_online_cpus();
+	cpus_read_lock();
 
 	data->nb.notifier_call = cpufreq_passive_notifier_call;
 	ret = cpufreq_register_notifier(&data->nb,
@@ -353,7 +353,7 @@ static int cpufreq_passive_register(struct devfreq_passive_data **p_data)
 	}
 
 out:
-	put_online_cpus();
+	cpus_read_unlock();
 	if (ret)
 		return ret;
 
