diff --git a/arch/arm64/boot/dts/mediatek/mt8183.dtsi b/arch/arm64/boot/dts/mediatek/mt8183.dtsi
index 91237e7..8b2c61f 100644
--- a/arch/arm64/boot/dts/mediatek/mt8183.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8183.dtsi
@@ -1675,6 +1675,24 @@ mmsys: syscon@14000000 {
 			mediatek,gce-client-reg = <&gce SUBSYS_1400XXXX 0 0x1000>;
 		};
 
+		mdp_camin@14000000 {
+                        compatible = "mediatek,mt8183-mdp-dl";
+                        mediatek,mdp-id = <0>;
+                        reg = <0 0x14000000 0 0x1000>;
+                        mediatek,gce-client-reg = <&gce SUBSYS_1400XXXX 0 0x1000>;
+                        clocks = <&mmsys CLK_MM_MDP_DL_TXCK>,
+                                <&mmsys CLK_MM_MDP_DL_RX>;
+                };
+
+                mdp_camin2@14000000 {
+                        compatible = "mediatek,mt8183-mdp-dl";
+                        mediatek,mdp-id = <1>;
+                        reg = <0 0x14000000 0 0x1000>;
+                        mediatek,gce-client-reg = <&gce SUBSYS_1400XXXX 0 0x1000>;
+                        clocks = <&mmsys CLK_MM_IPU_DL_TXCK>,
+                                <&mmsys CLK_MM_IPU_DL_RX>;
+                };
+
 		mdp3_rdma0: mdp3_rdma0@14001000 {
 			compatible = "mediatek,mt8183-mdp3",
 				     "mediatek,mt8183-mdp3-rdma";
@@ -1715,6 +1733,18 @@ mdp3_rdma0: mdp3_rdma0@14001000 {
 				     <&gce 0x15020000 SUBSYS_1502XXXX>;
 		};
 
+		mdp_imgi@15020000 {
+                        compatible = "mediatek,mt8183-mdp-imgi";
+                        mediatek,mdp-id = <0>;
+                        reg = <0 0x15020000 0 0x1000>;
+                        mediatek,gce-client-reg = <&gce SUBSYS_1502XXXX 0 0x1000>;
+                };
+
+                mdp_img2o@15020000 {
+                        compatible = "mediatek,mt8183-mdp-exto";
+                        mediatek,mdp-id = <1>;
+                };
+
 		mdp3_rsz0: mdp3_rsz0@14003000 {
 			compatible = "mediatek,mt8183-mdp3-rsz";
 			mediatek,mdp3-id = <0>;
