diff --git a/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c b/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c
index 49daea4..497bdf0 100644
--- a/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c
+++ b/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c
@@ -820,7 +820,7 @@ static int mtk_seninf_media_register(struct mtk_seninf *priv)
 		s_asd->port = vep.base.port;
 		s_asd->lanes = vep.bus.mipi_csi2.num_data_lanes;
 		ret = v4l2_async_notifier_add_fwnode_remote_subdev(
-				&priv->notifier, ep, &s_asd->asd);
+				&priv->notifier, ep, struct sensor_async_subdev);
 		if (ret)
 			goto err_parse;
 
