diff --git a/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c b/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c
index c5883dc..49daea4 100644
--- a/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c
+++ b/drivers/media/platform/mtk-isp/seninf/mtk_seninf.c
@@ -470,7 +470,7 @@ static void init_fmt(struct mtk_seninf *priv)
 }
 
 static int seninf_init_cfg(struct v4l2_subdev *sd,
-			   struct v4l2_subdev_pad_config *cfg)
+			   struct v4l2_subdev_state *cfg)
 {
 	struct v4l2_mbus_framefmt *mf;
 	unsigned int i;
@@ -484,7 +484,7 @@ static int seninf_init_cfg(struct v4l2_subdev *sd,
 }
 
 static int seninf_set_fmt(struct v4l2_subdev *sd,
-			  struct v4l2_subdev_pad_config *cfg,
+			  struct v4l2_subdev_state *cfg,
 			  struct v4l2_subdev_format *fmt)
 {
 	struct mtk_seninf *priv = sd_to_mtk_seninf(sd);
@@ -505,7 +505,7 @@ static int seninf_set_fmt(struct v4l2_subdev *sd,
 }
 
 static int seninf_get_fmt(struct v4l2_subdev *sd,
-			  struct v4l2_subdev_pad_config *cfg,
+			  struct v4l2_subdev_state *cfg,
 			  struct v4l2_subdev_format *fmt)
 {
 	struct mtk_seninf *priv = sd_to_mtk_seninf(sd);
@@ -519,7 +519,7 @@ static int seninf_get_fmt(struct v4l2_subdev *sd,
 }
 
 static int seninf_enum_mbus_code(struct v4l2_subdev *sd,
-				 struct v4l2_subdev_pad_config *cfg,
+				 struct v4l2_subdev_state *cfg,
 				 struct v4l2_subdev_mbus_code_enum *code)
 {
 	struct mtk_seninf *priv = sd_to_mtk_seninf(sd);
diff --git a/drivers/media/platform/mtk-isp/isp_50/dip/mtk_dip-v4l2.c b/drivers/media/platform/mtk-isp/isp_50/dip/mtk_dip-v4l2.c
index 2fcbbcc..1aa7ae6 100644
--- a/drivers/media/platform/mtk-isp/isp_50/dip/mtk_dip-v4l2.c
+++ b/drivers/media/platform/mtk-isp/isp_50/dip/mtk_dip-v4l2.c
@@ -61,7 +61,7 @@ static int mtk_dip_subdev_s_stream(struct v4l2_subdev *sd,
 }
 
 static int mtk_dip_subdev_get_fmt(struct v4l2_subdev *sd,
-				  struct v4l2_subdev_pad_config *cfg,
+				  struct v4l2_subdev_state *cfg,
 				  struct v4l2_subdev_format *fmt)
 {
 	struct mtk_dip_pipe *dip_pipe = mtk_dip_subdev_to_pipe(sd);
@@ -82,7 +82,7 @@ static int mtk_dip_subdev_get_fmt(struct v4l2_subdev *sd,
 }
 
 static int mtk_dip_subdev_set_fmt(struct v4l2_subdev *sd,
-				  struct v4l2_subdev_pad_config *cfg,
+				  struct v4l2_subdev_state *cfg,
 				  struct v4l2_subdev_format *fmt)
 {
 	struct mtk_dip_pipe *dip_pipe = mtk_dip_subdev_to_pipe(sd);
@@ -114,7 +114,7 @@ static int mtk_dip_subdev_set_fmt(struct v4l2_subdev *sd,
 }
 
 static int mtk_dip_subdev_get_selection(struct v4l2_subdev *sd,
-					struct v4l2_subdev_pad_config *cfg,
+					struct v4l2_subdev_state *cfg,
 					struct v4l2_subdev_selection *sel)
 {
 	struct v4l2_rect *try_sel, *r;
@@ -150,7 +150,7 @@ static int mtk_dip_subdev_get_selection(struct v4l2_subdev *sd,
 }
 
 static int mtk_dip_subdev_set_selection(struct v4l2_subdev *sd,
-					struct v4l2_subdev_pad_config *cfg,
+					struct v4l2_subdev_state *cfg,
 					struct v4l2_subdev_selection *sel)
 {
 	struct v4l2_rect *rect, *try_sel;
