diff --git a/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi b/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi
index a150dec..7fbbf95 100644
--- a/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8183-kukui.dtsi
@@ -794,6 +816,12 @@ pins2 {
 	};
 
 	scp_pins: scp {
+		pins_eint {
+                        pinmux = <PINMUX_GPIO5__FUNC_TP_GPIO5_AO>,
+                                 <PINMUX_GPIO6__FUNC_TP_GPIO6_AO>,
+                                 <PINMUX_GPIO7__FUNC_TP_GPIO7_AO>;
+                };
+
 		pins_scp_uart {
 			pinmux = <PINMUX_GPIO110__FUNC_TP_URXD1_AO>,
 				 <PINMUX_GPIO112__FUNC_TP_UTXD1_AO>;
