diff --git a/libfprint/drivers/vfs0090.c b/libfprint/drivers/vfs0090.c
index b0df52e..4b28d5e 100644
--- a/libfprint/drivers/vfs0090.c
+++ b/libfprint/drivers/vfs0090.c
@@ -2066,7 +2066,7 @@ static void dev_close(struct fp_img_dev *idev)
 /* Usb id table of device */
 static const struct usb_id id_table[] = {
 	{.vendor = 0x138a,.product = 0x0090},
-	{.vendor = 0x138a,.product = 0x0097},
+	{.vendor = 0x06cb,.product = 0x009a},
 	{0, 0, 0,},
 };
 
diff --git a/libfprint/drivers/vfs0090.h b/libfprint/drivers/vfs0090.h
index 0d7df78..9b0e9db 100644
--- a/libfprint/drivers/vfs0090.h
+++ b/libfprint/drivers/vfs0090.h
@@ -249,7 +249,7 @@ static const unsigned char INIT_SEQUENCE_RSP5[] = {
 	0x00, 0x80, 0x00, 0x00, 0x06, 0x06, 0x03, 0x00,
 	0x00, 0x80, 0x04, 0x00, 0x00, 0x80, 0x00, 0x00,
 	0x04, 0x03, 0x05, 0x00, 0x00, 0x00, 0x05, 0x00,
-	0x00, 0x00, 0x03, 0x00
+	0x00, 0x00, 0x08, 0x00
 };
 
 static const unsigned char INIT_SEQUENCE_MSG6[] = {
@@ -271,12 +271,14 @@ static const struct data_exchange_t INIT_SEQUENCES[] = {
 		.rsp_length = 68,
 	},
 	{
+		.weak_match = TRUE,
 		.msg = INIT_SEQUENCE_MSG3,
 		.msg_length = G_N_ELEMENTS(INIT_SEQUENCE_MSG3),
 		.rsp = NULL,
 		.rsp_length = 84,
 	},
 	{
+		.
 		.msg = INIT_SEQUENCE_MSG4,
 		.msg_length = G_N_ELEMENTS(INIT_SEQUENCE_MSG4),
 		.rsp = NULL,
@@ -427,8 +429,8 @@ static const unsigned char ACTIVATE_SEQUENCE_MSG345[] = {
 	0x75
 };
 static const unsigned char ACTIVATE_SEQUENCE_RSP345[] = {
-	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00,
-	0x71, 0x00                       /* 0x01 */
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x5a, 0x00,
+	0x90, 0x01                       /* 0x01 */
 };
 
 static const unsigned char ACTIVATE_SEQUENCE_MSG67[] = {
@@ -648,7 +650,7 @@ static const struct data_exchange_t ACTIVATE_SEQUENCES[] = {
 	{
 		.msg = ACTIVATE_SEQUENCE_MSG2,
 		.msg_length = G_N_ELEMENTS(ACTIVATE_SEQUENCE_MSG2),
-		.rsp = (unsigned char []) { 0x00, 0x00, 0x02, 0x00, 0x00, 0x00 },
+		.rsp = (unsigned char []) { 0x00, 0x00, 0x03, 0x00, 0x00, 0x00 },
 		.rsp_length = 6,
 	},
 	{
@@ -673,12 +675,14 @@ static const struct data_exchange_t ACTIVATE_SEQUENCES[] = {
 		.rsp_length = G_N_ELEMENTS(ACTIVATE_SEQUENCE_RSP345),
 	},
 	{
+		.weak_match = TRUE,
 		.msg = ACTIVATE_SEQUENCE_MSG67,
 		.msg_length = G_N_ELEMENTS(ACTIVATE_SEQUENCE_MSG67),
 		.rsp = NULL,
 		.rsp_length = 84,
 	},
 	{
+		.weak_match = TRUE,
 		.msg = ACTIVATE_SEQUENCE_MSG67,
 		.msg_length = G_N_ELEMENTS(ACTIVATE_SEQUENCE_MSG67),
 		.rsp = NULL,
