diff --git a/libfprint/fprint-list-udev-rules.c b/libfprint/fprint-list-udev-rules.c
index 5f17b6b..907b57d 100644
--- a/libfprint/fprint-list-udev-rules.c
+++ b/libfprint/fprint-list-udev-rules.c
@@ -30,6 +30,7 @@ static const FpIdEntry whitelist_id_table[] = {
   { .vid = 0x138a, .pid = 0x0091 },
   { .vid = 0x138a, .pid = 0x0094 },
   { .vid = 0x138a, .pid = 0x0097 },   /* Found on e.g. Lenovo T470s */
+  { .vid = 0x06cb, .pid = 0x009a },   /* Found on e.g. Lenovo X1 Gen6 */
   { .vid = 0 },
 };
 
