diff --git a/meson.build b/meson.build
index 036cc8d..9519c41 100644
--- a/meson.build
+++ b/meson.build
@@ -150,12 +150,13 @@ endif
 have_egl_xlib = have_egl and have_x11
 
 have_gles2 = get_option('gles2')
-if have_gles2
+have_native_backend = get_option('native_backend')
+if have_gles2 or have_native_backend
   gles2_dep = dependency('glesv2')
   gles2_libname = get_option('gles2_libname')
 
   if not have_egl
-    error('GLESv2 support requires EGL to be enabled')
+    error('GLESv2 and native backend support requires EGL to be enabled')
   endif
 endif
 
@@ -177,7 +178,6 @@ if have_libgudev
   gudev_dep = dependency('gudev-1.0', version: gudev_req)
 endif
 
-have_native_backend = get_option('native_backend')
 if have_native_backend
   libdrm_dep = dependency('libdrm')
   libgbm_dep = dependency('gbm', version: gbm_req)
@@ -194,10 +194,6 @@ if have_native_backend
     error('The native backend requires EGL to be enabled')
   endif
 
-  if not have_gles2
-    error('The native backend requires GLESv2 to be enabled')
-  endif
-
   if not have_libgudev
     error('The native backend requires udev to be enabled')
   endif
diff --git a/src/meson.build b/src/meson.build
index 90d8073..05f6ac7 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -36,7 +36,7 @@ if have_gl
   ]
 endif
 
-if have_gles2
+if have_native_backend
   mutter_pkg_private_deps += [
     gles2_dep,
   ]
