diff --git a/src/proximity.c b/src/proximity.c
index 0beca69..aa844e6 100644
--- a/src/proximity.c
+++ b/src/proximity.c
@@ -14,6 +14,7 @@
 #include "shell.h"
 #include "sensor-proxy-manager.h"
 #include "util.h"
+#include <unistd.h>
 
 /**
  * SECTION:proximity
@@ -150,6 +151,7 @@ on_proximity_near_changed (PhoshProximity          *self,
   gboolean near;
   int ps_mode;
   PhoshShell *shell = phosh_shell_get_default ();
+  PhoshWayland *wl = phosh_wayland_get_default ();
   PhoshMonitor *monitor = phosh_shell_get_builtin_monitor (shell);  
   
   near = phosh_dbus_sensor_proxy_get_proximity_near (
@@ -157,8 +159,15 @@ on_proximity_near_changed (PhoshProximity          *self,
 
   g_debug ("Proximity near changed: %d", near);
   if (near) {
-      ps_mode = PHOSH_MONITOR_POWER_SAVE_MODE_OFF;
+      if (!self->fader) {
+        self->fader = phosh_fader_new (phosh_wayland_get_zwlr_layer_shell_v1 (wl),
+                                       monitor->wl_output);
+        gtk_widget_show (GTK_WIDGET (self->fader));
+      }
+      sleep(5);
+      ps_mode = PHOSH_MONITOR_POWER_SAVE_MODE_OFF;    
   } else {
+      g_clear_pointer (&self->fader, phosh_cp_widget_destroy);
       ps_mode = PHOSH_MONITOR_POWER_SAVE_MODE_ON;
   }
   phosh_monitor_set_power_save_mode (monitor, ps_mode);
