diff --git a/src/meson.build b/src/meson.build
index 04e2558..8e2d82f 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -142,6 +142,8 @@ libphosh_sources = [
   'proximity.c',
   'sensor-proxy-manager.c',
   'sensor-proxy-manager.h',
+  'flashlightinfo.c',
+  'flashlightinfo.h',
   'rotateinfo.c',
   'rotateinfo.h',
   'screen-saver-manager.c',
diff --git a/src/settings.c b/src/settings.c
index 62271b5..eb9314e 100644
--- a/src/settings.c
+++ b/src/settings.c
@@ -23,9 +23,11 @@
 #include "rotateinfo.h"
 #include "feedbackinfo.h"
 #include "feedback-manager.h"
+#include "flashlightinfo.h"
 #include "notifications/notify-manager.h"
 #include "notifications/notification-frame.h"
 #include "media-player.h"
+#include <stdbool.h>
 
 #include <pulse/pulseaudio.h>
 #include "gvc-mixer-control.h"
@@ -69,6 +71,9 @@ typedef struct _PhoshSettings
   GtkWidget *list_notifications;
   GtkWidget *sw_notifications;
   LfbEvent  *notify_event;
+  
+  /* Flashlight */
+  GtkWidget *flashlightinfo;    
 } PhoshSettings;
 
 
@@ -112,6 +117,26 @@ feedback_setting_clicked_cb (PhoshSettings *self)
   phosh_feedback_manager_toggle (manager);
 }
 
+static void
+flashlight_setting_clicked_cb (PhoshSettings *self)
+{
+  bool currently_on = get_flashlight_status() == 1;
+  bool currently_off = get_flashlight_status() == 0;
+
+  if (currently_on)
+  {
+    phosh_status_icon_set_icon_name(PHOSH_STATUS_ICON (self->flashlightinfo), "torch-disabled-symbolic");
+    phosh_status_icon_set_info (PHOSH_STATUS_ICON (self->flashlightinfo), "Off");
+    set_flashlight_status(0);
+  }
+  if (currently_off)
+  {
+    phosh_status_icon_set_icon_name(PHOSH_STATUS_ICON (self->flashlightinfo), "torch-enabled-symbolic");
+    phosh_status_icon_set_info (PHOSH_STATUS_ICON (self->flashlightinfo), "On");
+    set_flashlight_status(1);
+  }
+}
+
 static void
 wifi_setting_clicked_cb (PhoshSettings *self)
 {
@@ -444,7 +469,6 @@ phosh_settings_finalize (GObject *object)
 }
 
 
-
 static void
 phosh_settings_class_init (PhoshSettingsClass *klass)
 {
@@ -465,6 +489,7 @@ phosh_settings_class_init (PhoshSettingsClass *klass)
   g_type_ensure (PHOSH_TYPE_BT_INFO);
   g_type_ensure (PHOSH_TYPE_DOCKED_INFO);
   g_type_ensure (PHOSH_TYPE_FEEDBACK_INFO);
+  g_type_ensure (PHOSH_TYPE_FLASHLIGHT_INFO);  
   g_type_ensure (PHOSH_TYPE_MEDIA_PLAYER);
   g_type_ensure (PHOSH_TYPE_QUICK_SETTING);
   g_type_ensure (PHOSH_TYPE_ROTATE_INFO);
@@ -475,6 +500,7 @@ phosh_settings_class_init (PhoshSettingsClass *klass)
   gtk_widget_class_bind_template_child (widget_class, PhoshSettings, quick_settings);
   gtk_widget_class_bind_template_child (widget_class, PhoshSettings, scale_brightness);
   gtk_widget_class_bind_template_child (widget_class, PhoshSettings, sw_notifications);
+  gtk_widget_class_bind_template_child (widget_class, PhoshSettings, flashlightinfo);
 
   gtk_widget_class_bind_template_callback (widget_class, battery_setting_clicked_cb);
   gtk_widget_class_bind_template_callback (widget_class, bt_setting_clicked_cb);
@@ -483,6 +509,7 @@ phosh_settings_class_init (PhoshSettingsClass *klass)
   gtk_widget_class_bind_template_callback (widget_class, feedback_setting_long_pressed_cb);
   gtk_widget_class_bind_template_callback (widget_class, on_media_player_raised);
   gtk_widget_class_bind_template_callback (widget_class, rotation_setting_clicked_cb);
+  gtk_widget_class_bind_template_callback (widget_class, flashlight_setting_clicked_cb);
   gtk_widget_class_bind_template_callback (widget_class, torch_setting_clicked_cb);
   gtk_widget_class_bind_template_callback (widget_class, wifi_setting_clicked_cb);
   gtk_widget_class_bind_template_callback (widget_class, wwan_setting_clicked_cb);
diff --git a/src/ui/settings-menu.ui b/src/ui/settings-menu.ui
index 7e9df0e..9874af7 100644
--- a/src/ui/settings-menu.ui
+++ b/src/ui/settings-menu.ui
@@ -120,13 +120,13 @@
                   </object>
                 </child>
                 <child>
-                  <object class="PhoshQuickSetting" id="torch_quick_setting">
+                  <object class="PhoshQuickSetting">
                     <property name="visible">True</property>
-                    <property name="sensitive" bind-source="torchinfo" bind-property="present" bind-flags="sync-create"/>
                     <property name="can_focus">False</property>
-                    <signal name="clicked" handler="torch_setting_clicked_cb" object="PhoshSettings" swapped="yes"/>
+                    <signal name="clicked" handler="flashlight_setting_clicked_cb" object="PhoshSettings" swapped="yes"/>
+
                     <child>
-                      <object class="PhoshTorchInfo" id="torchinfo">
+                      <object class="PhoshFlashlightInfo" id="flashlightinfo">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
                         <property name="icon-size">GTK_ICON_SIZE_LARGE_TOOLBAR</property>
